docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,379 @@
+# Rework of Mesh*Routes matching
+
+* Status: accepted
+
+Technical story: https://github.com/kumahq/kuma/issues/10247
+
+## Context and Problem Statement
+
+At the moment when targeting Mesh*Route you need to put it in topLevel targetRef. 
+We want to iterate over policy design to fully utilize the fact that we are referencing real objects.
+The current implementation has made things a little tricky because policy matching is convoluted where the top level 
+MeshHTTPRoute gets converted to a fake `to`.
+
+## Considered options
+
+- add possibility to put Mesh*Routes in `spec.to[].targetRef`
```

Comment:
maybe I am missing something here, but the more I think about it, that the two options are similar. Mesh*Route in `spec.to[].targetRef` gives more flexibility with overriding conf but `spec.targetRef` could be simpler to understand

Reply:
Right I was thinking more like:

> The spec.to[].targetRef field defines rules that applies to outgoing traffic of proxies selected by spec.targetRef. The spec.from[].targetRef field defines rules that applies to incoming traffic of proxies selected by spec.targetRef.

Clearly this hasn't been updated since we introduced MeshHTTPRoute top level. But can we update it now?

Previously `spec.to[].targetRef` defined which traffic was affected by selecting which destinations are affected. This isn't _quite_ true anymore, at least not directly.
How do we explain this? Should we define this in terms of Envoy resources?

Reply:
updated